### PR TITLE
Feat/5 wph model

### LINKED ISSUE #5: Implement wavelet phase harm model

### PR DESCRIPTION
**Pull request by amvs**
# Pull Request Summary

This pull request implements the **Wavelet Phase Harmonic (WPH) model architecture** and integrates it into the training pipeline. It introduces a modular layer system for wavelet convolutions, correlation computation, and phase-harmonic non-linearities, along with comprehensive testing and configuration infrastructure. Closes #5 

#### Core WPH Model Architecture

The core implementation is centered around `wph_model.py` and a new `layers/` module.

* **Model Definition:** Implemented `WPHModel` as the primary feature extractor and `WPHClassifier` as a wrapper class for classification tasks.
* **Modular Layers:** Decomposed the architecture into reusable `nn.Module` components:
    * `WaveConvLayer`: Handles wavelet convolutions (`layers/wave_conv_layer.py`).
    * `CorrLayer`: efficiently computes correlations (`layers/corr_layer.py`).
    * `HighpassLayer` & `LowpassLayer`: Handle frequency-specific filtering (`layers/highpass_layer.py`, `layers/lowpass_layer.py`).
    * `ReluCenterLayer`: Manages activation and centering operations (`layers/relu_center_layer.py`).
* **Backend & Utilities:**
    * Added `ALPHATorch` (`alpha_torch.py`) for specific WPH operations.
    * Implemented filter generation logic in `ops/build-filters.py` (including rotation in the Fourier domain).
    * Added initialization utilities `SubInitSpatialMean` and `DivInitStd` in `ops/backend.py`.

#### Testing

Added unit tests to ensure the mathematical correctness of the new architecture:

* Added specific tests for `CorrLayer`, `ReluCenterLayer`, and the overall `WPHModel`.
* Verified forward passes and gradient flows for the new layer implementations.

#### Training & Experimentation

* **Training Script:** Introduced specific training routines for the WPH model structure.
* **Configurations:** Added base configs and sweep configs for **MNIST** and **CUReT** datasets (`mnist.yaml`, `curet.yaml`, etc.) to enable systematic hyperparameter tuning.

#### Sweep & Infrastructure Updates

* **Sweep Script:** Enhanced `scripts/sweep.py` to support:
    * Model selection (`wph` vs `scat`).
    * Dynamic metric selection for sorting top models.
    * Merging and saving configuration files per run.
* **Checkpoint Cleanup:** Added `scripts/cleanup_checkpoints.py` to automatically prune sweep directories, retaining only the top-N models per validation ratio.

#### Logging & Utilities

* **Tensor Debugging:** Updated `LoggerManager` to log complex tensor states (checking for conjugation and complexity).
* **Robustness:** Added logging for uncaught exceptions and improved type safety for regularization parameters in `configs/config.py`.

### CODE REVIEW

File: configs/config.py [hunk 1]
```diff
@@ -39,6 +39,12 @@ def apply_overrides(config, overrides):
                 except ValueError:
                     pass  # leave as string
         config[k] = v
+    try:
```
Comment (yangl1n) [inline configs/config.py]: If overrides contain the configuration reseting for key 'lambda_reg', then it should already been converted to float.
If overrides not contain it but config contains it, it is the caller's responsibility to ensure that the value of 'lambda_reg' is valid.
Just don't understand why this piece of code is necessary.
